Sources/ProjectDescription/Dependency.swift

```diff
@@ -26,22 +26,47 @@ public enum Dependency: Codable, Equatable {
         case revision(String)
     }
 
+    /// Requirement for the Swift Package Manager dependency.
+    public enum SPMRequirement: Codable, Equatable {
+        /// Mimics `.exact("1.2.3")` from `Package.swift`.
+        case exact(Version)
+        /// Mimics `.upToNextMajor(from: "1.2.3")` from `Package.swift`.
+        case upToNextMajor(Version)
+        /// Mimics `.upToNextMinor(from: "1.2.3")` from `Package.swift`.
+        case upToNextMinor(Version)
+        /// Mimics `from: "1.2.3"` from `Package.swift`.
+        case atLeast(Version)
+        /// Mimics `"1.2.3"..<"1.2.6"` from `Package.swift`.
+        case range(Range<Version>)
+        /// Mimics `1.2.3"..."1.2.6` from `Package.swift`.`
+        case closedRange(ClosedRange<Version>)
+        /// Mimics `.branch("develop")` from `Package.swift`.
+        case branch(String)
+        /// Mimics `.revision("revision")` from `Package.swift`.
+        case revision(String)
+    }
+
     /// Contains the description of dependency that can by installed using Carthage. More: https://github.com/Carthage/Carthage/blob/master/Documentation/Artifacts.md
     case carthage(origin: CarthageOrigin, requirement: CarthageRequirement, platforms: Set<Platform>)
+    /// Contains the description of dependency that can by installed using Swift Package Manager. More: https://docs.swift.org/package-manager/PackageDescription/index.html
+    case spm(url: String, requirement: SPMRequirement)
```

Comment:
I think it's better to call it `package`. You'll also probably need to differentiate between a remote and local SPM dependency - I think your current solution allows only remote packages.

One way to approach this would be to add `remotePackage` and `localPackage` cases or add a `type` parameter to `package` case